### PR TITLE
Allele specific analysis

### PR DESCRIPTION
**Pull request by vivekbhr**
I am opening this PR here so that we can observe that other parallel changes from your side don't create a merge conflict..

Allele-specific RNA and DNA mapping is working.. ChIP-Seq peak calling and DE gene detection remains.

### CODE REVIEW

**Comment by vivekbhr** [comment]
After adding changes from master, I'll have to re-test this now with my files.. In the meanwhile can you check if my changes don't break your workflows..

I have tried to keep NO change in outputs, but many internal functions have changes. Especially review : 

[x] Introduced `--mode` in DNA-mapping.

[x] Moved all deeptools functions in RNA-seq and DNA-mapping to common `deeptools_cmds.snakemake`, functions are now shared between normal/allele-specific workflows. Some functions are also shared between DNA-mapping and RNA-seq workflow

[x] Snakefile `Bowtie2_bam_filtering` is now split into two snakefiles `Bowtie2` and `bam_filtering`

**Comment by vivekbhr** [comment]
Also closes #30

File: shared/deeptools_cmds.snakefile [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+                    "-o {output} " +
+                    "--labels {params.labels} " +
+                    "--binSize 1000 " +
+                    "{params.blacklist} " +
+                    "-p {threads} " +
+                    "{params.read_extension} " +
+                    "&> {log}") )
+
+# multiBAMsum RNA
+def multiBWsum_bed_cmd():
```
Comment (mirax87) [inline shared/deeptools_cmds.snakefile]: ambiguous  function name, suggesting to replace 'Sum' by 'Summary'

File: shared/deeptools_cmds.snakefile [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        "--BED {input.gtf} {input.gtf2} "
+        "--plotFile {output.png} "
+        "--labels {params.labels} "
+        "--plotTitle 'Fraction of reads in regions' "
+        "--outRawCounts {output.tsv} "
+        "--variableScales "
+        "&> {log}") )
+
+# multiBAMsum ChIP
+def multiBamSum_cmd():
```
Comment (mirax87) [inline shared/deeptools_cmds.snakefile]: line 69: ambiguous function name, suggesting to replace 'Sum' by 'Summary'

Comment (vivekbhr) [inline shared/deeptools_cmds.snakefile]: ambiguous with?

**Comment by vivekbhr** [comment]
by the way.. the cluster config change (my last commit) is not implemented in DNA-seq workflow. @steffenheyne is it only supposed to be for RNAseq?

**Comment by steffenheyne** [comment]
@vivekbhr cluster config:
My thinking so far was to make a workflow specific cluster config and only where necessary. 
So far, only RNA-seq needs it for some salmon steps 12G mem-per-cpu instead of the default 6G (set by SlurmEasy)

Does DNA-mapping now also needs it? Do you know the specific rules that need more memory?

**Comment by vivekbhr** [comment]
@steffenheyne for allele-specific pipeline the splitting of BAM files into allelic bam files requires >6G memory (it's single threaded). This would apply to both DNA and RNA mapping..

**Comment by vivekbhr** [comment]
by the way i am adding another check point here : the `/path/to/SNPsplit` in `paths.yaml`should be somewhere under `/package/`

- [ ] change path to SNPsplit in `paths.yaml`

File: shared/deeptools_cmds.snakefile (hunk 2 above)
Comment (vivekbhr) [inline shared/deeptools_cmds.snakefile]: done

**Comment by vivekbhr** [comment]
added chr X,Y,M for '--ignoreforNormalization'

File: shared/deeptools_cmds.snakefile [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+# plot PCA (both)
+def plotPCA_cmd(what):
+    return( (deepTools_path+"plotPCA " +
+            "-in {input} " +
+            "-o {output} " +
+            "-T 'PCA of fragment coverage' " +
+            "&> {log}") )
+
+# plot Coverage
+def plotCov_cmd():
```
Comment (vivekbhr) [inline shared/deeptools_cmds.snakefile]: done

**Comment by vivekbhr** [comment]
@mirax87 bamCoverage works with blacklist (tested OK)..

**Comment by vivekbhr** [comment]
Ready to merge..

**Comment by steffenheyne** [comment]
the "--ignoreForNormalization Y X M" feature might not be that easy to do it right...

Is it possible that you remove this functionality  from this PR and put it in a separate feature branch? this would make it easier to merge as we have less global changes that effect also all other workflows

**Pull request merged**